server/controllers/finance/debtors/index.js

```diff
@@ -280,20 +280,28 @@ function financialPatient(debtorUuid) {
     SELECT transaction.trans_id, BUID(transaction.entity_uuid) as entity_uuid, transaction.description,
       BUID(transaction.record_uuid) AS record_uuid, transaction.trans_date,
       SUM(transaction.credit_equiv) AS credit, SUM(transaction.debit_equiv) AS debit,
-      IF(ISNULL(transaction.reference), '', CONCAT_WS('.', '${identifiers.INVOICE.key}', transaction.abbr, transaction.reference)) AS reference
+      IF(ISNULL(transaction.reference), '', CONCAT_WS('.', '${identifiers.INVOICE.key}', transaction.abbr, transaction.reference)) AS reference, 
```

Comment:
You can just use the `entity_map` and `document_map` tables and avoid all these joins, I think.  That's what the table is there for.

Something like this (compressed for brevity):
```sql
SELECT transaction.trans_id, entity_map.text AS entityReference, document_map.text AS documentReference
  FROM transaction JOIN entity_map ON transaction.entity_uuid = entity_map.uuid
  JOIN document_map ON transaction.record_uuid = document_map.uuid;
```